core/src/main/java/org/geysermc/geyser/entity/type/ItemFrameEntity.java

```diff
@@ -114,15 +115,21 @@ public void setItemInFrame(EntityMetadata<ItemStack, ?> entityMetadata) {
         if (entityMetadata.getValue() != null) {
             this.heldItem = entityMetadata.getValue();
             ItemData itemData = ItemTranslator.translateToBedrock(session, heldItem);
+
             ItemMapping mapping = session.getItemMappings().getMapping(entityMetadata.getValue());
+            String identifier = session.getGeyser().customManager().getItemManager().itemStringFromId(itemData.getId());
```

Comment:
If it's not done already, is there any way we can place this inside of the session's `ItemMappings`?

Reply:
Camotoy said to keep custom model data items out of the ItemMappings, as they need to be handled like the original item. Fully custom items are in there though.

Reply:
I don't want custom items in the ItemMapping[] array because they don't have a unique Java ID associated with them. Custom item data in the ItemMappings class is fine.

Reply:
> If it's not done already, is there any way we can place this inside of the session's `ItemMappings`?
Looking at how I can do this, it looks like it's going to make the registration process even more complicated and slow. If this _needs_ to be done, it's possible though.

> I don't want custom items in the ItemMapping[] array because they don't have a unique Java ID associated with them. Custom item data in the ItemMappings class is fine.
I misunderstood then.

Reply:
Are you able to elaborate a bit more on this, or draft something simple up? If it's not worth the effort to continue then probably best to leave as-is, but just want to gauge what it would look like since imho it's probably better to have the mapping-related code here.